### PR TITLE
ANC Logic - Prenatal Diagnosis, record and store + Imminent Delivery diagnosis

### PR DESCRIPTION
**Pull request by anvmn**
#161 
#156

### CODE REVIEW

**Comment by anvmn** [comment]
Peek 2022-02-14 13-12

**Comment by anvmn** [comment]
@mariano-dagostino please review

File: client/src/elm/Backend/Update.elm [hunk 1]
```diff
@@ -1648,6 +1655,19 @@ updateIndexedDb language currentDate currentTime zscores nurseId healthCenterId
                     , extraMsgs
                     )
 
+                [ MedicationRevision uuid data ] ->
+                    let
+                        _ =
+                            Debug.log "" "MedicationRevision"
```
Comment (mariano-dagostino) [inline client/src/elm/Backend/Update.elm]: @anvmn Can this be removed?

File: client/src/elm/Backend/Update.elm [hunk 2]
```diff
@@ -3846,27 +3877,71 @@ generatePrenatalAssesmentMsgs currentDate language isChw updateAssesment after i
...
+                    urgentDiagnosis =
+                        EverySet.toList diagnosisAfter
+                            |> List.filter ((/=) DiagnosisPrescribeMebendezole)
+
+                    updateDiagnosisMsg =
+                        Backend.PrenatalEncounter.Model.SetPrenatalDiagnosis diagnosisAfter
+                            |> Backend.Model.MsgPrenatalEncounter id
+                            |> App.Model.MsgIndexedDb
+
+                    -- @todo
```
Comment (mariano-dagostino) [inline client/src/elm/Backend/Update.elm]: @anvmn What is pending here?

File: client/src/elm/Pages/PrenatalActivity/Utils.elm [hunk 3]
```diff
@@ -787,9 +898,31 @@ nextStepsTasksCompletedFromTotal language assembled data task =
...
+                --                     if isJust form.reasonForNotSendingToHC then
+                --                         ( 2, 2 )
+                --
+                --                     else
+                --                         ( 1, 2 )
+                --
+                --                 else
+                --                     ( 1, 1 )
+                --             )
+                --         |> Maybe.withDefault ( 0, 1 )
```
Comment (mariano-dagostino) [inline client/src/elm/Pages/PrenatalActivity/Utils.elm]: Can this be removed?

File: client/src/elm/Translate.elm [hunk 4]
```diff
@@ -3184,16 +3188,31 @@ translationSet trans =
...
+            , kinyarwanda = Nothing
+            }
+
+        DangerSignsHelperReferToHC ->
             { english = "Refer patient to health center immediately"
             , kinyarwanda = Just "Ibimenyetso Mpuruza"
             }
 
+        DangerSignsHelperReferToMaternityWard ->
+            { english = "Refer to Maternity Ward Immediately "
```
Comment (mariano-dagostino) [inline client/src/elm/Translate.elm]: Extra end space here

**Review by mariano-dagostino — COMMENTED** [review]
@anvmn Finished my review.

File: client/src/elm/Backend/Update.elm (hunk 2 above)
Comment (anvmn) [reply client/src/elm/Backend/Update.elm]: Added comment

File: server/hedley/modules/custom/hedley_restful/plugins/restful/node/HedleyRestfulPrenatalEncounter.class.php [hunk 5]
```diff
@@ -35,14 +35,14 @@ protected function alterQueryForViewWithDbSelect(SelectQuery $query) {
...
-      'field_prenatal_diagnosis',
+      'field_prenatal_diagnoses',
     ];
 
     foreach ($field_names as $field_name) {
       hedley_general_join_field_to_query($query, 'node', $field_name, FALSE);
     }
 
-    $query->addExpression("GROUP_CONCAT(DISTINCT field_prenatal_diagnosis.field_prenatal_diagnosis_value)", 'field_prenatal_diagnosis');
+    $query->addExpression("GROUP_CONCAT(DISTINCT field_prenatal_diagnoses.field_prenatal_diagnoses_value)", 'field_prenatal_diagnoses');
```
Comment (mariano-dagostino) [inline server/hedley/modules/custom/hedley_restful/plugins/restful/node/HedleyRestfulPrenatalEncounter.class.php:45]: @anvmn you probably did a full search and replace here, is this column present in the database? Or we should use the original value and keep the alias?

**Review by mariano-dagostino — APPROVED** [review]
@anvmn looks good, just  a question regarding the database query.

File: server/hedley/modules/custom/hedley_restful/plugins/restful/node/HedleyRestfulPrenatalEncounter.class.php (hunk 5 above)
Comment (anvmn) [reply server/hedley/modules/custom/hedley_restful/plugins/restful/node/HedleyRestfulPrenatalEncounter.class.php:45]: @mariano-dagostino 
Field name itself was changed.
As this development is new, and was not deployed yet, column at DB will be created with new name.

**Pull request merged**